src/app/shared/services/vm-snapshot.service.ts

```diff
@@ -43,10 +53,9 @@ export class VmSnapshotService extends BaseBackendService<VmSnapshot> {
       );
   }
 
-  public revert(id: string): Observable<any> {
-    // todo it returns vm, maybe we need to update vms
+  public revert(id: string): Observable<VirtualMachine> {
     return super
       .sendCommand('revertTo', { vmsnapshotid: id })
-      .pipe(switchMap(job => this.asyncJobService.queryJob(job, this.entity)));
+      .pipe(switchMap(job => this.asyncJobService.queryJob(job, 'virtualmachine')));
```

Comment:
Why do you use `this.entity` in `create` and `virtualmachine` in `revert`?

Reply:
Async service resolves data coming from API based on the second parameter of `queryJob` function. Create action returns VM snapshot, but revert action return VM.